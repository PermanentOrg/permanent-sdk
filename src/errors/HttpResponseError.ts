export class HttpResponseError extends Error {
  public readonly statusCode: number;

  public constructor(statusCode: number, message: string) {
    super(message);
    this.statusCode = statusCode;
  }
}
