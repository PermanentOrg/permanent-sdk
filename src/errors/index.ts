export * from './HttpResponseError';
export * from './ValidationError';
