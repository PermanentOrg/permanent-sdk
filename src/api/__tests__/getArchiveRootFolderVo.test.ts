import path from "node:path";
import nock from "nock";
import { getArchiveRootFolderVo } from "..";
import { ValidationError } from "../../errors";

describe("getArchiveRootFolderVo", () => {
	it("should return the root folder VO", async () => {
		nock("https://permanent.local")
			.get("/api/folder/getRoot")
			.query({ archiveId: 1 })
			.replyWithFile(
				200,
				path.join(
					__dirname,
					"fixtures",
					"getArchiveRootFolderVo",
					"archiveRoot.json",
				),
				{
					"Content-Type": "application/json",
				},
			);

		const archiveVos = await getArchiveRootFolderVo(
			{
				bearerToken: "12345",
				baseUrl: "https://permanent.local/api",
			},
			1,
		);

		expect(archiveVos).toMatchSnapshot();
	});

	it("should error when provided an invalid response", async () => {
		nock("https://permanent.local")
			.get("/api/folder/getRoot")
			.query({ archiveId: 1 })
			.reply(200, '["Not at all a folderVo"]', {
				"Content-Type": "application/json",
			});

		await expect(
			getArchiveRootFolderVo(
				{
					bearerToken: "12345",
					baseUrl: "https://permanent.local/api",
				},
				1,
			),
		).rejects.toBeInstanceOf(ValidationError);
	});
});
