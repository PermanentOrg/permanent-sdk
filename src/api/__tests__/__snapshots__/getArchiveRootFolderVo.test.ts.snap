// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`getArchiveRootFolderVo should return the root folder VO 1`] = `
{
  "AccessVO": null,
  "AccessVOs": null,
  "ChildFolderVOs": [],
  "ChildItemVOs": [
    {
      "AccessVO": null,
      "AccessVOs": null,
      "ChildFolderVOs": [],
      "ChildItemVOs": [],
      "FolderLinkVOs": [
        {
          "accessRole": "access.role.owner",
          "archiveId": 1,
          "createdDT": "2022-12-16T15:45:10",
          "folderId": 2,
          "folder_linkId": 2,
          "linkCount": 1,
          "parentFolderId": 1,
          "parentFolder_linkId": 1,
          "position": 1,
          "recordId": null,
          "shareDT": null,
          "status": "status.generic.ok",
          "type": "type.folder_link.root.app",
          "updatedDT": "2022-12-16T15:45:10",
        },
      ],
      "FolderSizeVO": null,
      "LocnVO": null,
      "ParentFolderVOs": [],
      "RecordVOs": [],
      "ShareVOs": [],
      "SharedArchiveVOs": [],
      "TagVOs": [],
      "TimezoneVO": {
        "country": null,
        "countryCode": null,
        "createdDT": "2022-12-16T15:45:10",
        "displayName": "Central Time",
        "dstAbbrev": "CDT",
        "dstName": null,
        "dstOffset": "-05:00",
        "status": "status.generic.ok",
        "stdAbbrev": "CST",
        "stdName": null,
        "stdOffset": "-06:00",
        "timeZoneId": 88,
        "timeZonePlace": null,
        "type": "type.folder.root.app",
        "updatedDT": "2022-12-16T15:45:11",
      },
      "accessRole": "access.role.owner",
      "archiveArchiveNbr": null,
      "archiveId": 1,
      "archiveNbr": "0001-0002",
      "createdDT": "2022-12-16T15:45:10",
      "derivedDT": null,
      "derivedEndDT": null,
      "description": "Welcome to apps section. This is where you can manage all the 3rd party apps you have connected with.",
      "displayDT": "2022-12-16T15:45:10",
      "displayEndDT": null,
      "displayName": "Apps",
      "downloadName": "Apps",
      "downloadNameOk": null,
      "folderId": 2,
      "folder_linkId": 2,
      "folder_linkType": "type.folder_link.root.app",
      "imageRatio": "1.00",
      "locnId": null,
      "note": null,
      "parentArchiveNbr": null,
      "parentDisplayName": null,
      "parentFolderId": 1,
      "parentFolder_linkId": 1,
      "pathAsArchiveNbr": [],
      "pathAsFolder_linkId": null,
      "pathAsText": [],
      "posLimit": null,
      "posStart": null,
      "position": 1,
      "publicDT": null,
      "returnDataSize": null,
      "searchScore": null,
      "shareDT": null,
      "sort": null,
      "special": null,
      "status": "status.generic.ok",
      "thumbArchiveNbr": null,
      "thumbDT": "2023-06-16T15:45:10",
      "thumbStatus": "status.generic.ok",
      "thumbURL1000": "https://localcdn.permanent.org/_slifty/0001-0002.thumb.w200?t=1686926710&Expires=1686926710&Signature=Q9fSp~pEalhYehcnQuL-1LB8YCwpnZrc7Vx66~U5HpcLthPNaoimUP3X4AsCQILBFk07woq3oUws-EUbUuxJovdsFPNbsUZzz64zIv-kO4Ck2nlgCnEtLyNkfhajf5VvTFaymTzpNVN1oMyOQNMwg~NEvL18sqJkPjTWh4NkF6Ace9wAhjK5d23q2v4Onxbfm50uDHojyWisI-2PNG4~65f7ZGGTIp9vE3wZGvUcXn8KJR2FUCP6hnaq33EiMhkJxzHAWIKNrKsd08HVXnH3EID3ry1EdbbBnucYebDbCk8kZQj4u1n~PI5pBql1YmVK-QCwtPpeyf8-2AE-VbXCMw__&Key-Pair-Id=APKAJP2D34UGZ6IG443Q",
      "thumbURL200": "https://localcdn.permanent.org/_slifty/0001-0002.thumb.w200?t=1686926710&Expires=1686926710&Signature=Q9fSp~pEalhYehcnQuL-1LB8YCwpnZrc7Vx66~U5HpcLthPNaoimUP3X4AsCQILBFk07woq3oUws-EUbUuxJovdsFPNbsUZzz64zIv-kO4Ck2nlgCnEtLyNkfhajf5VvTFaymTzpNVN1oMyOQNMwg~NEvL18sqJkPjTWh4NkF6Ace9wAhjK5d23q2v4Onxbfm50uDHojyWisI-2PNG4~65f7ZGGTIp9vE3wZGvUcXn8KJR2FUCP6hnaq33EiMhkJxzHAWIKNrKsd08HVXnH3EID3ry1EdbbBnucYebDbCk8kZQj4u1n~PI5pBql1YmVK-QCwtPpeyf8-2AE-VbXCMw__&Key-Pair-Id=APKAJP2D34UGZ6IG443Q",
      "thumbURL2000": "https://localcdn.permanent.org/_slifty/0001-0002.thumb.w200?t=1686926710&Expires=1686926710&Signature=Q9fSp~pEalhYehcnQuL-1LB8YCwpnZrc7Vx66~U5HpcLthPNaoimUP3X4AsCQILBFk07woq3oUws-EUbUuxJovdsFPNbsUZzz64zIv-kO4Ck2nlgCnEtLyNkfhajf5VvTFaymTzpNVN1oMyOQNMwg~NEvL18sqJkPjTWh4NkF6Ace9wAhjK5d23q2v4Onxbfm50uDHojyWisI-2PNG4~65f7ZGGTIp9vE3wZGvUcXn8KJR2FUCP6hnaq33EiMhkJxzHAWIKNrKsd08HVXnH3EID3ry1EdbbBnucYebDbCk8kZQj4u1n~PI5pBql1YmVK-QCwtPpeyf8-2AE-VbXCMw__&Key-Pair-Id=APKAJP2D34UGZ6IG443Q",
      "thumbURL500": "https://localcdn.permanent.org/_slifty/0001-0002.thumb.w200?t=1686926710&Expires=1686926710&Signature=Q9fSp~pEalhYehcnQuL-1LB8YCwpnZrc7Vx66~U5HpcLthPNaoimUP3X4AsCQILBFk07woq3oUws-EUbUuxJovdsFPNbsUZzz64zIv-kO4Ck2nlgCnEtLyNkfhajf5VvTFaymTzpNVN1oMyOQNMwg~NEvL18sqJkPjTWh4NkF6Ace9wAhjK5d23q2v4Onxbfm50uDHojyWisI-2PNG4~65f7ZGGTIp9vE3wZGvUcXn8KJR2FUCP6hnaq33EiMhkJxzHAWIKNrKsd08HVXnH3EID3ry1EdbbBnucYebDbCk8kZQj4u1n~PI5pBql1YmVK-QCwtPpeyf8-2AE-VbXCMw__&Key-Pair-Id=APKAJP2D34UGZ6IG443Q",
      "timeZoneId": 88,
      "type": "type.folder.root.app",
      "updatedDT": "2022-12-16T15:45:11",
      "view": "folder.view.grid",
      "viewProperty": null,
    },
    {
      "AccessVO": null,
      "AccessVOs": null,
      "ChildFolderVOs": [],
      "ChildItemVOs": [],
      "FolderLinkVOs": [
        {
          "accessRole": "access.role.owner",
          "archiveId": 1,
          "createdDT": "2022-12-16T15:45:11",
          "folderId": 3,
          "folder_linkId": 3,
          "linkCount": 1,
          "parentFolderId": 1,
          "parentFolder_linkId": 1,
          "position": 1,
          "recordId": null,
          "shareDT": null,
          "status": "status.generic.ok",
          "type": "type.folder_link.root.private",
          "updatedDT": "2022-12-16T15:45:11",
        },
      ],
      "FolderSizeVO": null,
      "LocnVO": null,
      "ParentFolderVOs": [],
      "RecordVOs": [],
      "ShareVOs": [],
      "SharedArchiveVOs": [],
      "TagVOs": [],
      "TimezoneVO": {
        "country": null,
        "countryCode": null,
        "createdDT": "2022-12-16T15:45:10",
        "displayName": "Central Time",
        "dstAbbrev": "CDT",
        "dstName": null,
        "dstOffset": "-05:00",
        "status": "status.generic.ok",
        "stdAbbrev": "CST",
        "stdName": null,
        "stdOffset": "-06:00",
        "timeZoneId": 88,
        "timeZonePlace": null,
        "type": "type.folder.root.private",
        "updatedDT": "2022-12-16T16:15:13",
      },
      "accessRole": "access.role.owner",
      "archiveArchiveNbr": null,
      "archiveId": 1,
      "archiveNbr": "0001-0003",
      "createdDT": "2022-12-16T15:45:10",
      "derivedDT": null,
      "derivedEndDT": null,
      "description": "Welcome to My Files, a place to organize all the files you have uploaded. If you would like to share something with someone, you can do it by selecting that item and choosing the desired user and their access permission.",
      "displayDT": "2022-12-16T15:45:10",
      "displayEndDT": null,
      "displayName": "My Files",
      "downloadName": "My Files",
      "downloadNameOk": null,
      "folderId": 3,
      "folder_linkId": 3,
      "folder_linkType": "type.folder_link.root.private",
      "imageRatio": "1.00",
      "locnId": null,
      "note": null,
      "parentArchiveNbr": null,
      "parentDisplayName": null,
      "parentFolderId": 1,
      "parentFolder_linkId": 1,
      "pathAsArchiveNbr": [],
      "pathAsFolder_linkId": null,
      "pathAsText": [],
      "posLimit": null,
      "posStart": null,
      "position": 1,
      "publicDT": null,
      "returnDataSize": null,
      "searchScore": null,
      "shareDT": null,
      "sort": null,
      "special": null,
      "status": "status.generic.ok",
      "thumbArchiveNbr": null,
      "thumbDT": "2023-06-16T15:45:11",
      "thumbStatus": "status.generic.ok",
      "thumbURL1000": "https://localcdn.permanent.org/_slifty/0001-0003.thumb.w200?t=1686926711&Expires=1686926711&Signature=NCjb5Q2cGrOIQRyxrfM-HFpsbLOFdt22yU1MZbPu0DbsvHhNi5dstYfYlEdnQXQ6RErN1utwmri~rl9-rV92ebRBYJB~gvLQ2SGmJTB9J5SoCK-~6PabtdXcL6Rnf~Th5q0WdKXp4T9GtXC67lyumhS3Seu7UAIH0XWjxLd941Y9bfc-Z3wQhXSG4wS3wbnFTSBjDSWQCdTQKSHU8bJcIaA2cAA5W2~TtEqUADV7diGhNU3urFyRgIhJcp6Yy0l5VdxTF7BdAOKlludFO53aj3CVkqY9k~esl2bW1ZkAZ-6mh1f8EvykE3fi61YJaz3m-I224GfaO6uAWcI95TDi3A__&Key-Pair-Id=APKAJP2D34UGZ6IG443Q",
      "thumbURL200": "https://localcdn.permanent.org/_slifty/0001-0003.thumb.w200?t=1686926711&Expires=1686926711&Signature=NCjb5Q2cGrOIQRyxrfM-HFpsbLOFdt22yU1MZbPu0DbsvHhNi5dstYfYlEdnQXQ6RErN1utwmri~rl9-rV92ebRBYJB~gvLQ2SGmJTB9J5SoCK-~6PabtdXcL6Rnf~Th5q0WdKXp4T9GtXC67lyumhS3Seu7UAIH0XWjxLd941Y9bfc-Z3wQhXSG4wS3wbnFTSBjDSWQCdTQKSHU8bJcIaA2cAA5W2~TtEqUADV7diGhNU3urFyRgIhJcp6Yy0l5VdxTF7BdAOKlludFO53aj3CVkqY9k~esl2bW1ZkAZ-6mh1f8EvykE3fi61YJaz3m-I224GfaO6uAWcI95TDi3A__&Key-Pair-Id=APKAJP2D34UGZ6IG443Q",
      "thumbURL2000": "https://localcdn.permanent.org/_slifty/0001-0003.thumb.w200?t=1686926711&Expires=1686926711&Signature=NCjb5Q2cGrOIQRyxrfM-HFpsbLOFdt22yU1MZbPu0DbsvHhNi5dstYfYlEdnQXQ6RErN1utwmri~rl9-rV92ebRBYJB~gvLQ2SGmJTB9J5SoCK-~6PabtdXcL6Rnf~Th5q0WdKXp4T9GtXC67lyumhS3Seu7UAIH0XWjxLd941Y9bfc-Z3wQhXSG4wS3wbnFTSBjDSWQCdTQKSHU8bJcIaA2cAA5W2~TtEqUADV7diGhNU3urFyRgIhJcp6Yy0l5VdxTF7BdAOKlludFO53aj3CVkqY9k~esl2bW1ZkAZ-6mh1f8EvykE3fi61YJaz3m-I224GfaO6uAWcI95TDi3A__&Key-Pair-Id=APKAJP2D34UGZ6IG443Q",
      "thumbURL500": "https://localcdn.permanent.org/_slifty/0001-0003.thumb.w200?t=1686926711&Expires=1686926711&Signature=NCjb5Q2cGrOIQRyxrfM-HFpsbLOFdt22yU1MZbPu0DbsvHhNi5dstYfYlEdnQXQ6RErN1utwmri~rl9-rV92ebRBYJB~gvLQ2SGmJTB9J5SoCK-~6PabtdXcL6Rnf~Th5q0WdKXp4T9GtXC67lyumhS3Seu7UAIH0XWjxLd941Y9bfc-Z3wQhXSG4wS3wbnFTSBjDSWQCdTQKSHU8bJcIaA2cAA5W2~TtEqUADV7diGhNU3urFyRgIhJcp6Yy0l5VdxTF7BdAOKlludFO53aj3CVkqY9k~esl2bW1ZkAZ-6mh1f8EvykE3fi61YJaz3m-I224GfaO6uAWcI95TDi3A__&Key-Pair-Id=APKAJP2D34UGZ6IG443Q",
      "timeZoneId": 88,
      "type": "type.folder.root.private",
      "updatedDT": "2022-12-16T16:15:13",
      "view": "folder.view.grid",
      "viewProperty": null,
    },
    {
      "AccessVO": null,
      "AccessVOs": null,
      "ChildFolderVOs": [],
      "ChildItemVOs": [],
      "FolderLinkVOs": [
        {
          "accessRole": "access.role.owner",
          "archiveId": 1,
          "createdDT": "2022-12-16T15:45:11",
          "folderId": 4,
          "folder_linkId": 4,
          "linkCount": 1,
          "parentFolderId": 1,
          "parentFolder_linkId": 1,
          "position": 1,
          "recordId": null,
          "shareDT": null,
          "status": "status.generic.ok",
          "type": "type.folder_link.root.public",
          "updatedDT": "2022-12-16T15:45:11",
        },
      ],
      "FolderSizeVO": null,
      "LocnVO": null,
      "ParentFolderVOs": [],
      "RecordVOs": [],
      "ShareVOs": [],
      "SharedArchiveVOs": [],
      "TagVOs": [],
      "TimezoneVO": {
        "country": null,
        "countryCode": null,
        "createdDT": "2022-12-16T15:45:11",
        "displayName": "Central Time",
        "dstAbbrev": "CDT",
        "dstName": null,
        "dstOffset": "-05:00",
        "status": "status.generic.ok",
        "stdAbbrev": "CST",
        "stdName": null,
        "stdOffset": "-06:00",
        "timeZoneId": 88,
        "timeZonePlace": null,
        "type": "type.folder.root.public",
        "updatedDT": "2022-12-16T15:45:13",
      },
      "accessRole": "access.role.owner",
      "archiveArchiveNbr": null,
      "archiveId": 1,
      "archiveNbr": "0001-0004",
      "createdDT": "2022-12-16T15:45:11",
      "derivedDT": null,
      "derivedEndDT": null,
      "description": "This is your Public folder. Place files and information in this folder that you would want everyone in the world to be able to see for eternity. Do not put information here that you do not want publicly available right away. Tell your life story. Add the things you would like to share and preserve.",
      "displayDT": "2022-12-16T15:45:11",
      "displayEndDT": null,
      "displayName": "Public",
      "downloadName": "Public",
      "downloadNameOk": null,
      "folderId": 4,
      "folder_linkId": 4,
      "folder_linkType": "type.folder_link.root.public",
      "imageRatio": "1.00",
      "locnId": null,
      "note": null,
      "parentArchiveNbr": null,
      "parentDisplayName": null,
      "parentFolderId": 1,
      "parentFolder_linkId": 1,
      "pathAsArchiveNbr": [],
      "pathAsFolder_linkId": null,
      "pathAsText": [],
      "posLimit": null,
      "posStart": null,
      "position": 1,
      "publicDT": "2022-12-15T00:00:00",
      "returnDataSize": null,
      "searchScore": null,
      "shareDT": null,
      "sort": null,
      "special": null,
      "status": "status.generic.ok",
      "thumbArchiveNbr": null,
      "thumbDT": "2023-12-16T15:45:12",
      "thumbStatus": "status.generic.ok",
      "thumbURL1000": "https://localcdn.permanent.org/_slifty/0001-0004.thumb.w200?t=1702741512&Expires=1702741512&Signature=Hemsa9vnAJuExBXJwpbFOIb9RRKS3CL7j0gTwLYBnE8FRpwNlAuqvqKl4DnrK~Z7SyBHTkoDxT8JjbzQ3aDtftQW0VoKBAcbRdM9yXRBc2DxOTq-cSe-78UOefJjtktQIdPME0m054dJz-QfaGHdWnNuhINiDYCQequz9Kp4m1R1U6hUkOs1y-z0TN4HlKb~E6R2xJuE-TSKrgCUCw5JFGkDHAqccBl~xbLDdcy9swZ95k4MA5c-Hg-mYutLplIUCoW1PhV1bnvnNshu-j5AVZ3~Y0F8cdfn4vmciwoUdvEUT8YmnVz4A7sOWYoUDnGE7Ml1kCPf0oQFBn-ieHMUDg__&Key-Pair-Id=APKAJP2D34UGZ6IG443Q",
      "thumbURL200": "https://localcdn.permanent.org/_slifty/0001-0004.thumb.w200?t=1702741512&Expires=1702741512&Signature=Hemsa9vnAJuExBXJwpbFOIb9RRKS3CL7j0gTwLYBnE8FRpwNlAuqvqKl4DnrK~Z7SyBHTkoDxT8JjbzQ3aDtftQW0VoKBAcbRdM9yXRBc2DxOTq-cSe-78UOefJjtktQIdPME0m054dJz-QfaGHdWnNuhINiDYCQequz9Kp4m1R1U6hUkOs1y-z0TN4HlKb~E6R2xJuE-TSKrgCUCw5JFGkDHAqccBl~xbLDdcy9swZ95k4MA5c-Hg-mYutLplIUCoW1PhV1bnvnNshu-j5AVZ3~Y0F8cdfn4vmciwoUdvEUT8YmnVz4A7sOWYoUDnGE7Ml1kCPf0oQFBn-ieHMUDg__&Key-Pair-Id=APKAJP2D34UGZ6IG443Q",
      "thumbURL2000": "https://localcdn.permanent.org/_slifty/0001-0004.thumb.w200?t=1702741512&Expires=1702741512&Signature=Hemsa9vnAJuExBXJwpbFOIb9RRKS3CL7j0gTwLYBnE8FRpwNlAuqvqKl4DnrK~Z7SyBHTkoDxT8JjbzQ3aDtftQW0VoKBAcbRdM9yXRBc2DxOTq-cSe-78UOefJjtktQIdPME0m054dJz-QfaGHdWnNuhINiDYCQequz9Kp4m1R1U6hUkOs1y-z0TN4HlKb~E6R2xJuE-TSKrgCUCw5JFGkDHAqccBl~xbLDdcy9swZ95k4MA5c-Hg-mYutLplIUCoW1PhV1bnvnNshu-j5AVZ3~Y0F8cdfn4vmciwoUdvEUT8YmnVz4A7sOWYoUDnGE7Ml1kCPf0oQFBn-ieHMUDg__&Key-Pair-Id=APKAJP2D34UGZ6IG443Q",
      "thumbURL500": "https://localcdn.permanent.org/_slifty/0001-0004.thumb.w200?t=1702741512&Expires=1702741512&Signature=Hemsa9vnAJuExBXJwpbFOIb9RRKS3CL7j0gTwLYBnE8FRpwNlAuqvqKl4DnrK~Z7SyBHTkoDxT8JjbzQ3aDtftQW0VoKBAcbRdM9yXRBc2DxOTq-cSe-78UOefJjtktQIdPME0m054dJz-QfaGHdWnNuhINiDYCQequz9Kp4m1R1U6hUkOs1y-z0TN4HlKb~E6R2xJuE-TSKrgCUCw5JFGkDHAqccBl~xbLDdcy9swZ95k4MA5c-Hg-mYutLplIUCoW1PhV1bnvnNshu-j5AVZ3~Y0F8cdfn4vmciwoUdvEUT8YmnVz4A7sOWYoUDnGE7Ml1kCPf0oQFBn-ieHMUDg__&Key-Pair-Id=APKAJP2D34UGZ6IG443Q",
      "timeZoneId": 88,
      "type": "type.folder.root.public",
      "updatedDT": "2022-12-16T15:45:13",
      "view": "folder.view.grid",
      "viewProperty": null,
    },
  ],
  "FolderLinkVOs": [
    {
      "accessRole": "access.role.owner",
      "archiveId": 1,
      "createdDT": "2022-12-16T15:45:08",
      "folderId": 1,
      "folder_linkId": 1,
      "linkCount": 1,
      "parentFolderId": null,
      "parentFolder_linkId": null,
      "position": 1,
      "recordId": null,
      "shareDT": null,
      "status": "status.generic.ok",
      "type": "type.folder_link.root.root",
      "updatedDT": "2022-12-16T15:45:08",
    },
  ],
  "FolderSizeVO": null,
  "LocnVO": null,
  "ParentFolderVOs": [],
  "RecordVOs": [],
  "ShareVOs": [],
  "SharedArchiveVOs": [],
  "TagVOs": [],
  "TimezoneVO": null,
  "accessRole": "access.role.owner",
  "archiveArchiveNbr": null,
  "archiveId": 1,
  "archiveNbr": "0001-0001",
  "createdDT": "2022-12-16T15:45:08",
  "derivedDT": "2022-12-16T15:45:08",
  "derivedEndDT": null,
  "description": null,
  "displayDT": "2022-12-16T15:45:08",
  "displayEndDT": null,
  "displayName": "Archive Root",
  "downloadName": "Archive Root",
  "downloadNameOk": true,
  "folderId": 1,
  "folder_linkId": 1,
  "folder_linkType": "type.folder_link.root.root",
  "imageRatio": "1.00",
  "locnId": null,
  "note": null,
  "parentArchiveNbr": null,
  "parentDisplayName": null,
  "parentFolderId": null,
  "parentFolder_linkId": null,
  "pathAsArchiveNbr": [],
  "pathAsFolder_linkId": null,
  "pathAsText": [],
  "posLimit": null,
  "posStart": null,
  "position": 1,
  "publicDT": null,
  "returnDataSize": null,
  "searchScore": null,
  "shareDT": null,
  "sort": "sort.alphabetical_asc",
  "special": null,
  "status": "status.generic.ok",
  "thumbArchiveNbr": null,
  "thumbDT": "2023-06-16T15:45:09",
  "thumbStatus": "status.generic.ok",
  "thumbURL1000": "https://localcdn.permanent.org/_slifty/0001-0001.thumb.w200?t=1686926709&Expires=1686926709&Signature=eFODdrMXyIsnROXKCDcVUTEjlAmmw99-KcvJW-YCyJ9xbH32gNLKFmCxwzWuA2oHRAxvi-qUcx9Wluq3FxZKDRMWzj~IRruGjsfJ3YpsLI854dajtew~mnNBSQOu9Bpy9dFKizuoTfW0UF8ZebqhVa6UHRaNuTl0zHQh0eqTG0HNzkKih4p-fvk2MnplRCsCAcdg5OYhXTeguhVBGfVFq8KC0VY34jWJ3GMIwCh~u1uFEauD2nhCfrAWVgIx6v5HSZqeg5rFcErNHMGAmOPI2UJyBZScZRFZ43ZQWjRelRrtJ~EKxj6l584-xJhw5c64oN9vTXeXZXR1JCE3VzuAoA__&Key-Pair-Id=APKAJP2D34UGZ6IG443Q",
  "thumbURL200": "https://localcdn.permanent.org/_slifty/0001-0001.thumb.w200?t=1686926709&Expires=1686926709&Signature=eFODdrMXyIsnROXKCDcVUTEjlAmmw99-KcvJW-YCyJ9xbH32gNLKFmCxwzWuA2oHRAxvi-qUcx9Wluq3FxZKDRMWzj~IRruGjsfJ3YpsLI854dajtew~mnNBSQOu9Bpy9dFKizuoTfW0UF8ZebqhVa6UHRaNuTl0zHQh0eqTG0HNzkKih4p-fvk2MnplRCsCAcdg5OYhXTeguhVBGfVFq8KC0VY34jWJ3GMIwCh~u1uFEauD2nhCfrAWVgIx6v5HSZqeg5rFcErNHMGAmOPI2UJyBZScZRFZ43ZQWjRelRrtJ~EKxj6l584-xJhw5c64oN9vTXeXZXR1JCE3VzuAoA__&Key-Pair-Id=APKAJP2D34UGZ6IG443Q",
  "thumbURL2000": "https://localcdn.permanent.org/_slifty/0001-0001.thumb.w200?t=1686926709&Expires=1686926709&Signature=eFODdrMXyIsnROXKCDcVUTEjlAmmw99-KcvJW-YCyJ9xbH32gNLKFmCxwzWuA2oHRAxvi-qUcx9Wluq3FxZKDRMWzj~IRruGjsfJ3YpsLI854dajtew~mnNBSQOu9Bpy9dFKizuoTfW0UF8ZebqhVa6UHRaNuTl0zHQh0eqTG0HNzkKih4p-fvk2MnplRCsCAcdg5OYhXTeguhVBGfVFq8KC0VY34jWJ3GMIwCh~u1uFEauD2nhCfrAWVgIx6v5HSZqeg5rFcErNHMGAmOPI2UJyBZScZRFZ43ZQWjRelRrtJ~EKxj6l584-xJhw5c64oN9vTXeXZXR1JCE3VzuAoA__&Key-Pair-Id=APKAJP2D34UGZ6IG443Q",
  "thumbURL500": "https://localcdn.permanent.org/_slifty/0001-0001.thumb.w200?t=1686926709&Expires=1686926709&Signature=eFODdrMXyIsnROXKCDcVUTEjlAmmw99-KcvJW-YCyJ9xbH32gNLKFmCxwzWuA2oHRAxvi-qUcx9Wluq3FxZKDRMWzj~IRruGjsfJ3YpsLI854dajtew~mnNBSQOu9Bpy9dFKizuoTfW0UF8ZebqhVa6UHRaNuTl0zHQh0eqTG0HNzkKih4p-fvk2MnplRCsCAcdg5OYhXTeguhVBGfVFq8KC0VY34jWJ3GMIwCh~u1uFEauD2nhCfrAWVgIx6v5HSZqeg5rFcErNHMGAmOPI2UJyBZScZRFZ43ZQWjRelRrtJ~EKxj6l584-xJhw5c64oN9vTXeXZXR1JCE3VzuAoA__&Key-Pair-Id=APKAJP2D34UGZ6IG443Q",
  "timeZoneId": 88,
  "type": "type.folder.root.root",
  "updatedDT": "2022-12-16T15:45:10",
  "view": "folder.view.grid",
  "viewProperty": null,
}
`;
